draft-ietf-quic-transport.md

```diff
@@ -770,6 +775,29 @@ are in a different packet number space to other packets (see
 {{packet-numbers}}).
 
 
+### 0-RTT Packet Numbers {#retry-0rtt-pn}
+
+Packet numbers for 0-RTT protected packets use the same space as 1-RTT protected
+packets.
+
+After a client receives a Retry or Version Negotiation packet, it MAY attempt to
+send data in 0-RTT packets after it sends a new Initial packet.  However, a
+client MUST NOT reset the packet number it uses for 0-RTT packets.  The keys
```

Comment:
If the CID doesn't change, reordering or caching could cause the first flight of 0-RTT packets to remain valid after the new Initial lands.  So unless the CID changes (and maybe even if it does, if the server is trying too hard), you have to assume those packets are still outstanding but potentially lost.  Maybe we should frame this in terms of fast retransmit?

Reply:
I realize now the source of the confusion.  You are concerned about the case where maybe all this 0-RTT is received (and can be processed).  I'm more concerned about catastrophic compromise of packet protection.  I think that the nonce reuse thing is more urgent.